test/Microsoft.Azure.ServiceBus.UnitTests/MessageInterop/MessageInteropEnd2EndTests.cs

```diff
@@ -0,0 +1,109 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#if NET46
+namespace Microsoft.Azure.ServiceBus.UnitTests.MessageInterop
+{
+    using Microsoft.Azure.ServiceBus.InteropExtensions;
+    using Microsoft.ServiceBus.Messaging;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Text;
+    using System.Threading.Tasks;
+    using Xunit;
+    using System;
+
+    public class MessageInteropEnd2EndTests
+    {
+        public static IEnumerable<object> TestEnd2EndEntityPermutations => new object[]
+        {
+            new object[] { TransportType.NetMessaging },
+            new object[] { TransportType.Amqp }
+        };
+
+        [Theory]
+        [MemberData(nameof(TestEnd2EndEntityPermutations))]
+        [DisplayTestMethodName]
+        async Task RunEnd2EndSerializerTests(TransportType transportType)
+        {
+            string queueName = TestConstants.NonPartitionedQueueName;
+
+            // Override and Create a new ConnectionString with SbmpConnection Endpoint scheme
+            string[] temp = TestUtility.NamespaceConnectionString.Split(':');
+            string sbConnectionString = "Endpoint=sb:" + temp[1];
+
+            if(transportType == TransportType.Amqp)
+            {
+                sbConnectionString += ';' + nameof(TransportType) + '=' + TransportType.Amqp.ToString();
```

Comment:
> ';' + nameof(TransportType) + '=' + TransportType.Amqp.ToString();

could be passed in as a parameter via permutations parameters, just like `transportType` is.